software/base/src/main/java/org/apache/brooklyn/entity/software/base/lifecycle/MachineLifecycleEffectorTasks.java

```diff
@@ -444,9 +493,6 @@ public void run() {
                 }
             }
             entity().addLocations(ImmutableList.of((Location) machine));
-            if (entity().getAttribute(Attributes.JCLOUDS_PROVISIONING_RUNNING) != null) {
```

Comment:
Why you do not call
```
entity().sensors().remove(PROVISIONING_TASK_STATE)
entity().sensors().remove(PROVISIONED_MACHINE)
```
Otherwise these sensors will stay after machine provisioning completes.